### PR TITLE
[Tests-Only] Refactor file test

### PR DESCRIPTION
**Pull request by HariBhandari07**
### Description
This PR transforms all tests in `filesTest.js` to pact.io.

### Related Issue
Related to https://github.com/owncloud/owncloud-sdk/issues/500

### CODE REVIEW

**Review by swoichha — COMMENTED** [review]
LGTM

File: tests/filesTest.js [hunk 1]
```diff
@@ -202,345 +337,362 @@ fdescribe('Main: Currently testing files management,', function () {
...
-      expect(error).toBe(null)
-      done()
+      oc.files.mkdir(newFolder).then(status => {
+        expect(status).toBe(true)
+        return oc.files.list(newFolder, 0)
+      }).then(folder => {
+        folder = folder[0]
+        expect(folder.isDir()).toBe(true)
+        expect(folder.getName()).toEqual('new folder')
+        return oc.files.delete(newFolder)
```
Comment (individual-it) [inline tests/filesTest.js:369]: no need to delete the folder after the test, this test is about mkdir and there is no real server running where the folder is created. so this "delete" test should be moved to a separate test, or delete it completely if there are other explicit delete tests (can be done in a separate PR)

File: tests/filesTest.js [hunk 2]
```diff
@@ -202,345 +337,362 @@ fdescribe('Main: Currently testing files management,', function () {
...
-    }).catch(error => {
-      expect(error.message).toBe('File with name ' + config.nonExistentDir.slice(1) + ' could not be located')
-      done()
+      oc.files.mkdir(newFolder).then(status => {
+        expect(status).toBe(true)
+        return oc.files.list(newFolder, 0)
+      }).then(folder => {
+        folder = folder[0]
+        expect(folder.isDir()).toBe(true)
+        expect(folder.getName()).toEqual('subdir')
```
Comment (individual-it) [inline tests/filesTest.js:419]: no need to create those folders, they are anyway not created for real

File: tests/filesTest.js [hunk 3]
```diff
@@ -202,345 +337,362 @@ fdescribe('Main: Currently testing files management,', function () {
...
+      }).then(status2 => {
+        expect(status2).toEqual(true)
+        return oc.files.list(newFolder, 0)
+      }).then(folder2 => {
+        fail(folder2)
+        done()
+      }).catch(error => {
+        expect(error.message).toBe('File with name ' + newFolder + ' could not be located')
+        done()
+      })
```
Comment (individual-it) [inline tests/filesTest.js:430]: no need to list those folders afterwards

**Review by individual-it — CHANGES_REQUESTED** [review]
looks generally good but:
1. write more comments about what you have done in the PR description (comments are love-letters to your future-self)
2. there are still setup and teardown jobs done (e.g. create a folder to test deletion and visa-verse) - we don't need them because we only run the tests on a mock server. make explicit mkdir and explicit delete tests

**Review by individual-it — APPROVED** [review]

**Pull request merged**